central/resourcecollection/datastore/datastore_impl.go

```diff
@@ -564,6 +590,16 @@ func verifyCollectionConstraints(collection *storage.ResourceCollection) error {
 				if _, err := regexp.Compile(ruleValue.GetValue()); err != nil {
```

Comment:
Is this check needed for `labelType` fields?

Reply:
Good catch, I changed it to only check the regex compilation when it's a regex match rule